src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -665,6 +672,650 @@ void M1()
+        Operand: 
+          IIsPatternOperation (OperationKind.IsPattern, Type: System.Boolean, IsInvalid) (Syntax: 'y is int z')
+            Value: 
+              ILocalReferenceOperation: y (OperationKind.LocalReference, Type: System.Int32, IsInvalid) (Syntax: 'y')
+            Pattern: 
+              IDeclarationPatternOperation (OperationKind.DeclarationPattern, Type: null, IsInvalid) (Syntax: 'int z') (InputType: System.Int32, DeclaredSymbol: System.Int32 z, MatchesNull: False)
+  Declarators:
+      IVariableDeclaratorOperation (Symbol: System.Int32[] x) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'x')
+        Initializer: 
+          null
+  Initializer: 
+    null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(6,13): warning CS0219: The variable 'y' is assigned but its value is never used
+                //         int y = 10;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "y").WithArguments("y").WithLocation(6, 13),
+                // file.cs(7,22): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //         /*<bind>*/int[y is int z] x;/*</bind>*/
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[y is int z]").WithLocation(7, 22),
+                // file.cs(7,23): error CS0029: Cannot implicitly convert type 'bool' to 'int'
+                //         /*<bind>*/int[y is int z] x;/*</bind>*/
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "y is int z").WithArguments("bool", "int").WithLocation(7, 23),
+                // file.cs(7,35): warning CS0168: The variable 'x' is declared but never used
+                //         /*<bind>*/int[y is int z] x;/*</bind>*/
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "x").WithArguments("x").WithLocation(7, 35)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IVariableDeclaration_InvalidIgnoredDimensions_SwitchExpression()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        int y = 10;
+        /*<bind>*/int[] x;/*</bind>*/
+    }
+}
+";
+            // We replace
+            // int[] x;
+            // with
+            // int[y switch { int z => 42 }] x;
+
+            var syntaxTree = Parse(source, filename: "file.cs");
+            var rankSpecifierOld = syntaxTree.GetCompilationUnitRoot().DescendantNodes().OfType<ArrayRankSpecifierSyntax>().Single();
+            var rankSpecifierNew = rankSpecifierOld
+                .WithSizes(SyntaxFactory.SeparatedList<ExpressionSyntax>(
+                    SyntaxFactory.NodeOrTokenList(
+                        SyntaxFactory.SwitchExpression(
```

Comment:
>SyntaxFactory.SwitchExpression( [](start = 24, length = 31)

Does SyntaxFactory have a ParseExpression helper? Might simplify getting a syntax tree for a switch expression.